client/components/mma/shared/asyncComponents/LoadingComponent.tsx

```diff
@@ -0,0 +1,32 @@
+import useAsyncLoader from '../../../../utilities/hooks/useAsyncLoader';
+import type ResponseProcessor from './ResponseProcessor';
+
+export type LoadingProps = {
+	asyncFetch: () => Promise<any>;
+	responseProcessor: ResponseProcessor;
+	LoadingView: () => React.ReactElement;
+	LoadedView: (data: any) => React.ReactElement;
+	ErrorView: () => React.ReactElement;
+};
+
+export function LoadingComponent({
+	asyncFetch,
+	responseProcessor,
+	LoadingView,
+	LoadedView,
```

Comment:
I still don't like this variable - too similar to LoadingView... but this is a hard thing to name.